### PR TITLE
LSE-31 If a user opens the About page of a course that has the visibility setting set to 'none', a page with no styling opens.

### PR DESCRIPTION
**Pull request by RuslanSkiraRaccoongang**
**Description:** Rewrite the middleware class _curl.CurrentRequestUserMiddleware_ and _process_exception_ method. The main idea use Custom theme in rendering 404.html page.

**Youtrack:** [LSE-31](https://youtrack.raccoongang.com/issue/LSE-31)

**Testing instructions:**

1. Open CMS -> any course->Advanced settings -> Visibility 'none'
2. Open LMS -> 'current course'/about/ 
3. Should be 404 page with current theme

**Reviewers:**
- [ ] @sendr

### CODE REVIEW

File: lms/djangoapps/courseware/middleware.py [hunk 1]
```diff
@@ -22,3 +24,11 @@ def process_exception(self, _request, exception):
...
                 )
             )
+
+class CustomCurrentRequestUserMiddleware(crum.CurrentRequestUserMiddleware):
+    """
+    Overide the process_exception() method in CurrentRequestUserMiddleware class for
+    using custom theme with 404 page.
+    """
+    def process_exception(self, request, exception):
+        pass
```
Comment (sendr) [inline lms/djangoapps/courseware/middleware.py]: please add the empty line

File: lms/djangoapps/courseware/middleware.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 
 from courseware.courses import UserNotEnrolled
 
+import crum
```
Comment (sendr) [inline lms/djangoapps/courseware/middleware.py]: incorrect import order

File: lms/djangoapps/courseware/middleware.py (hunk 1 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/courseware/middleware.py]: Done

File: lms/djangoapps/courseware/middleware.py (hunk 2 above)
Comment (RuslanSkiraRaccoongang) [reply lms/djangoapps/courseware/middleware.py]: Done

**Comment by sendr** [comment]
add please ticket title to the PR title

**Pull request merged**